scRNA-seq-advanced/setup/intro-citeseq-setup.Rmd

```diff
@@ -0,0 +1,410 @@
+Currently, the ADTs and genes are both part of the primary experiment, so let's move the ADT counts into an `altExp` slot.
+
+```{r}
+# We can see this in the `Type` column in the rowData() where the creators of this dataset
+#  stored experiement information
+unique(rowData(pbmc_sce_raw)$Type)
+
+# Split out the ADT into an altExp
+pbmc_sce_split <- splitAltExps(pbmc_sce_raw, 
+                               #"Gene Expression" or "Antibody Capture"
+                               rowData(pbmc_sce_raw)$Type, 
+                               # The Type which should be the main experiment
+                               "Gene Expression")
+
+
+# The names for the experiments were directly derived from `Type`:
+mainExpName(pbmc_sce_split)
+altExpNames(pbmc_sce_split)
+
+# But here on the computer, we don't like spaces, so let's rename it to something informative
+#  and space-free, like "CITEseq" or "ADT". We'll use "ADT" since typing less > typing more
+altExpNames(pbmc_sce_split) <- "ADT"
+
+# And confirm:
+altExpNames(pbmc_sce_split)
+
+# We can do this for main if we want too!
+#mainExpName(pbmc_sce_split) <- "RNA"
+```
+
+### Quick ADT exploration
+
+Let's look at our ADTs:
+
+```{r}
+# Here's how we can see the altExp itself - it's its own SCE object
+altExp(pbmc_sce_split)
+
+# What are the tags?
+rownames(altExp(pbmc_sce_split))
+```
+
+These are all human surface proteins except `IgG1`, which is a mouse isotype that was used as a control. 
+We'd, if there's no contamination, that there are 0 counts for this protein since the data is human:
+
+```{r}
+# Here's the counts assay:
+counts(altExp(pbmc_sce_split))
+
+# Use rowSums to see totals:
+# Wow, almost no IgG1! Hurray. 
+rowSums(counts(altExp(pbmc_sce_split)))
+```
+
+
+> At this point we could consider removing cells with IgG1 > 0 and remove the antibody entirely.
+
+### Filter RNA
+
+#### First, with `DropletUtils`
+
+```{r}
+# Filter RNA
+droplet_df <- DropletUtils::emptyDropsCellRanger(
+  counts(pbmc_sce_split), 
+  BPPARAM = BiocParallel::MulticoreParam(4)
+) 
+
+# Filter down to low FDR cells and check out dimensions
+# Values here are mostly NA.
+# NOTE: this filtering does propagate to altExp!! :) 
+cells_to_retain <- which(droplet_df$FDR <= 0.01)
+filtered_sce <- pbmc_sce_split[, cells_to_retain]
+dim(pbmc_sce_split)  # before filtering
+dim(filtered_sce)  # after filtering
+```
+
+
+#### Now, with `miQC`
+
+We'll need some mitochondria stats for `miQC`:
+
+```{r}
+filtered_sce <- scater::addPerCellQC(
+  filtered_sce, 
+  subsets = list(mito = mito_genes))
+
+
+# See that `sum` is RNA only and `total` is both RNA and ADT?
+head(filtered_sce$sum)
+head(filtered_sce$total)
+
+all(filtered_sce$total >= filtered_sce$sum) # total is never going to be LOWER than sum
+```
+
+```{r}
+miqc_model <- miQC::mixtureModel(filtered_sce) 
+filtered_sce <- miQC::filterCells(filtered_sce,
```

Comment:
I think we might to explore miQC a bit and look at some of the outputs. I think this is another PR, but just wanted to comment that I don't want to leave miQC at this simplicity and I think we should explore it a bit more.

Reply:
Agreed!